qiskit/tools/visualization.py

```diff
@@ -674,6 +679,60 @@ def plot_wigner_data(wigner_data, phis=None, method=None):
 ###############################################################
 
 
+def plot_circuit(circuit, basis="u1,u2,u3,cx,x,y,z,h,s,t,rx,ry,rz"):
+    """Plot and show circuit (opens new window, cannot inline in Jupyter)
+    Defaults to an overcomplete basis, in order to not alter gates.
+    Note: Requires pdflatex installed (to compile Latex)
+    Note: Requires poppler installed (to convert pdf to image)
+    """
+    im = circuit_drawer(circuit, basis)
+    if im:
+        im.show()
+
+
+def circuit_drawer(circuit, basis="u1,u2,u3,cx,x,y,z,h,s,t,rx,ry,rz"):
+    """Obtain the circuit in image format (output can be inlined in Jupyter)
+    Defaults to an overcomplete basis, in order to not alter gates.
+    Note: Requires pdflatex installed (to compile Latex)
+    Note: Requires poppler installed (to convert pdf to image)
+    """
+    filename = 'circuit'
+    tmpdir = 'tmp/'
```

Comment:
Can we use [`tempfile.TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#examples) or similar for creating the temporary directory? It's quite handy for these cases, as it will make sure that the folder is created in a safe place and also takes care of deleting it if using it as a context manager.

Reply:
Yes that's much better. Done!